utils/timed_version.go

```diff
@@ -0,0 +1,142 @@
+package utils
+
+import (
+	"fmt"
+	"sync/atomic"
```

Comment:
We have replaced used of this with https://pkg.go.dev/go.uber.org/atomic everywhere. Is it possible to use the uber atomic package here?

Reply:
i added nocmp below so it has the safety and ergonomics as a memeber of the uber atomics family. using that to implement it would make it messier though